index.bs

```diff
@@ -1462,3 +1485,278 @@ The [=remote end steps=] with |command parameters| are:
+    |document|'s <a spec=dom>URL</a>.
+
+    Note: This includes the fragment component of the URL.
+
+ 1. Let |child info| be the result of [=get the ancestor browsing contexts=]
+    given |context id|, |depth| + 1, and |max depth|.
+
+ 1. Let |context info| be a [=map=] matching the <code>BrowsingContextInfo</code>
+    production with the <code>context</code> field set to |context id|, the
+    <code>parent</code> field set to |parent id|, the <code>url</code> field
+    set to |url|, and the <code>children</code> field set to |child info|.
+
+ 1. Return |context info|.
+
+</div>
+
+<div algorithm>
+  To <dfn>get the ancestor browsing contexts</dfn> given |parent id|, |depth|
+  and |max depth|:
+
+  1. If |max depth| is greater than zero, and |depth| is equal to |max depth|,
+     return null.
+
+  1. Let |parent| be the result of [=trying=] to [=get a browsing context=]
+     given |parent id|.
+
+  1. If |parent| is null, let |child contexts| be a list containing all [=top-level
+     browsing contexts=]. Otherwise let |child contexts| be a list containing all
+     [=/browsing contexts=] which are [=child browsing contexts=] of |parent|.
+
+  1. Let |contexts info| be a list.
+
+  1. For each |context| of |child contexts|:
+
+    1. Let |info| be the result of [=get the browsing context info=] given
+       |context|, |depth|, and |max depth|.
+
+    1. Append |info| to |contexts info|
+
+  1. Return |contexts info|
+
+</div>
+
+### Commands ### {#module-browsingContext-commands}
+
+#### The browsingContext.getTree Command ####  {#command-browsingContext-list}
+
+The <dfn export for=commands>browsingContext.getTree</dfn> command returns a
+tree of all browsing contexts that are children of the given context, or all
+top-level contexts when no parent is provided.
+
+<dl>
+   <dt>Command Type</dt>
+   <dd>
+      <pre class="cddl remote-cddl">
+      BrowsingContextGetTreeCommand = {
+        method: "browsingContext.getTree",
+        params: BrowsingContextGetTreeParameters
+      }
+
+      BrowsingContextGetTreeParameters = {
+        ?maxDepth: uint,
+        ?parent: BrowsingContext,
+      }
+      </pre>
+   </dd>
+   <dt>Return Type</dt>
+   <dd>
+    <pre class="cddl local-cddl">
+        BrowsingContextGetTreeResult = {
+          contexts: [*BrowsingContextInfo]
+        }
+    </pre>
+   </dd>
+</dl>
+
+<div algorithm="remote end steps for browsingContext.getTree">
+The [=remote end steps=] with |command parameters| are:
+
+  1. Let the |parent id| be the value of the <code>parent</code> field of
+     |command parameters| if present, or null otherwise.
+
+  1. Let |max depth| be the value of the <code>maxDepth</code> field of |command
+     parameters| if present, or 0 otherwise.
+
+  1. Let |depth| be 0.
+
+  1. Let |contexts| be the result of [=get the ancestor browsing contexts=],
+     given |parent id|, |depth|, and |max depth|.
+
+  1. Let |body| be a [=map=] matching the <code>BrowsingContextListResult</code>
```

Comment:
Should it be BrowsingContextGetTreeResult instead of BrowsingContextListResult?

Reply:
Good catch.